backend/src/routes/v1/webAde.js

```diff
@@ -0,0 +1,211 @@
+  // If the user has "WEBADE_CFG_READ_ALL" then they can get all
+  let acronyms = [];
+  const roles = req.user.jwt.realm_access.roles;
+  let hasReadAllRole = false;
+  if (typeof roles === 'object' && roles instanceof Array) {
+    acronyms = utils.filterAppAcronymRoles(roles);
+    hasReadAllRole = roles.includes('WEBADE_CFG_READ_ALL');
+  }
+
+  if (!hasReadAllRole) {
+    const appAcronym = req.params.appAcronym;
+    if (!acronyms.includes(appAcronym)) {
+      return res.status(403).json({
+        message: `User lacks permission for '${appAcronym}' acronym`
+      });
+    }
+  }
+
+  try {
+    const response = await appConfigComponent.getAppConfigs(req.params.webAdeEnv);
+    if (response) {
+      return res.status(200).json(utils.filterWebAdeDependencies(response, req.params.appAcronym));
+    } else {
+      return res.status(404).json();
+    }
+  } catch (error) {
+    log.error(error);
+    res.status(500).json({
+      message: error.message
+    });
+    return res;
+  }
+});
+
+// fetches a list of all application Preferences from the webade config list that match a search criteria in the pref name
+// and have the secure indicator set to FALSE
+webAde.get('/:webAdeEnv/preferences/insecurePrefs', [
+], async (req, res) => {
+  var searchCriteria = req.query.searchCriteria;
+
+  // Check for required permissions.
+  // If the user has "WEBADE_CFG_READ_ALL" then they can get all
+  const roles = req.user.jwt.realm_access.roles;
+  let hasReadAllRole = false;
+  if (typeof roles === 'object' && roles instanceof Array) {
+    hasReadAllRole = roles.includes('WEBADE_CFG_READ_ALL');
+  }
+
+  if (!hasReadAllRole) {
+    return res.status(403).json({
+      message: 'User lacks permission to read all webade configs'
+    });
+  }
+
+  try {
+    const response = await appConfigComponent.getAppConfigs(req.params.webAdeEnv);
+    if (response) {
+      return res.status(200).json(utils.filterForInsecurePrefs(response, searchCriteria));
+    } else {
+      return res.status(404).json();
+    }
+  } catch (error) {
+    log.error(error);
+    res.status(500).json({
+      message: error.message
+    });
+    return res;
+  }
+});
+
+// gets the big array of all webade configs for a specified env
+webAde.get('/:webAdeEnv/appConfigs', [
+], async (req, res) => {
+  // Only administrative users with WEBADE_CFG_READ_ALL can get this info
+  const roles = req.user.jwt.realm_access.roles;
+  let hasReadAllRole = false;
+  if (typeof roles === 'object' && roles instanceof Array) {
+    hasReadAllRole = roles.includes('WEBADE_CFG_READ_ALL');
+  }
+
+  if (!hasReadAllRole) {
+    return res.status(403).json({
+      message: 'User lacks permission to get all app configs'
+    });
+  }
+
+  try {
+    const response = await appConfigComponent.getAppConfigs(req.params.webAdeEnv);
+    if (response) {
+      return res.status(200).json(response);
+    } else {
+      return res.status(404).json();
```

Comment:
If you're not intending on returning a body, it is better to use `.end()` instead of `.json()` without content.